basic/13-decentralized-exchange/uniswap-v1-like/test/Exchange.test.js

```diff
@@ -1,24 +1,30 @@
-require('@nomiclabs/hardhat-waffle');
-const { BigNumber, utils, provider } = ethers;
 const { expect } = require('chai');
+const hre = require('hardhat');
+const ethers = hre.ethers;
+
 const { getGasFeeFromTx } = require('../utils');
+const { getAddress } = require('ethers');
 
-const toWei = (value) => utils.parseEther(value.toString());
+const toWei = (value) => ethers.parseEther(value.toString());
 
-const fromWei = (value) => utils.formatEther(typeof value === 'string' ? value : value.toString());
+const fromWei = (value) =>
+  ethers.formatEther(typeof value === 'string' ? value : value.toString());
 
-const getBalance = provider.getBalance;
+const getBalance = async (value) =>
+  await ethers.provider.getBalance(value);
 
 const createExchange = async (factory, tokenAddress, sender) => {
-  const exchangeAddress = await factory.connect(sender).callStatic.createExchange(tokenAddress);
 
-  await factory.connect(sender).createExchange(tokenAddress);
+  console.log(tokenAddress, sender.address);
+  const exchangeAddress = await factory.connect(sender).createExchange(tokenAddress);
+  exchangeAddress.wait();
 
+  const address = await factory.connect(sender).getExchange(tokenAddress);
+  console.log("token address-" + tokenAddress + "-exchange address-" + address);
+  console.log(address);
```

Comment:
delete the log ?

Reply:
I won't delete the log statement console.log("token address-" + tokenAddress + "-exchange address-" + address); during my study. This log helps me debug the program and understand its principles.